src/metabase/api/metabot.clj

```diff
@@ -0,0 +1,95 @@
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+#_{:clj-kondo/ignore [:deprecated-var]}
+(api/defendpoint-schema POST "/model/:model-id"
+  "Ask Metabot to generate a SQL query given a prompt about a given model."
+  [model-id :as {{:keys [question]} :body}]
+  ;{model-id ms/PositiveInt
+  ; question string?}
+  (log/infof
+   "Metabot '/api/metabot/model/%s' being called with prompt: '%s'"
+   model-id
+   question)
+  (let [model   (api/check-404 (t2/select-one Card :id model-id :dataset true))
+        context {:model       (metabot-util/denormalize-model model)
+                 :user_prompt question
+                 :prompt_task :infer_sql}]
+    (or
+     (metabot/infer-sql context)
+     (throw
+      (let [message (format
+                     "Query '%s' didn't produce any SQL. Perhaps try a more detailed query."
+                     question)]
+        (ex-info
+         message
+         {:status-code 400
+          :message     message}))))))
+
+#_{:clj-kondo/ignore [:deprecated-var]}
+(api/defendpoint-schema POST "/database/:database-id"
+  "Ask Metabot to generate a SQL query given a prompt about a given database."
+  [database-id :as {{:keys [question]} :body}]
+  {database-id su/IntGreaterThanZero
+   question    su/NonBlankString}
+  (log/infof
+   "Metabot '/api/metabot/database/%s' being called with prompt: '%s'"
+   database-id
+   question)
+  (let [{:as database} (api/check-404 (t2/select-one Database :id database-id))
+        context {:database    (metabot-util/denormalize-database database)
+                 :user_prompt question
+                 :prompt_task :infer_model}]
+    (if-some [model (metabot/infer-model context)]
+      (or
+       (metabot/infer-sql (merge
+                           context
+                           {:model       model
+                            :prompt_task :infer_sql}))
+       (throw
+        (let [message (format
+                       "Query '%s' didn't produce any SQL. Perhaps try a more detailed query."
+                       question)]
+          (ex-info
+           message
+           {:status-code 400
+            :message     message}))))
+      (throw
+       (let [message (format
+                      (str/join
+                       " "
+                       ["Query '%s' didn't find a good match to your data."
+                        "Perhaps try a query that mentions the model name or columns more specifically."])
+                      question)]
+         (ex-info
+          message
+          {:status-code 400
+           :message     message}))))))
+
+#_{:clj-kondo/ignore [:deprecated-var]}
+(api/defendpoint-schema POST "/feedback"
+  "Record feedback on metabot results."
+  [:as {feedback :body}]
+  (if-some [stored-feedback (metabot-feedback/submit-feedback feedback)]
+    {:feedback stored-feedback
+     :message  "Thanks for your feedback"}
+    (throw
+     (let [message "There was a problem submitting your feedback."]
```

Comment:
do we care about 500s? I wonder if this should asynchronously fire off the feedback and then unequivocally say "thank you"

Reply:
I don't feel strongly about this. I agree that the end user probably just doesn't care. Fire and forget is ok.

Reply:
We're going to try to get this merged, so I'll add this as a TODO on the fast follow.